### ISSUE TITLE
False positive report of `UnusedVariable` for fields used with JUnit5 `FieldSource` annotation

### ISSUE THREAD

**Issue by genericjon** [google/error-prone#4713]
A variable that is used by JUnit5's `FieldSource` annotation will be flagged as an unused variable. The `FieldSource` annotation was introduced in JUnit version 5.11.

Example:
```java
class FieldSourceTest {
    private static final List<String> fooParameters = List.of("apple", "banana");

    @ParameterizedTest
    @FieldSource("fooParameters")
    public void foo(String bar) {
        System.out.println(bar);
    }
}
```
Output: 
```
FieldSourceTest.java:2: error: [UnusedVariable] The field 'fooParameters' is never read.
```
This is similar to https://github.com/google/error-prone/issues/2335, where methods used by `MethodSource` were flagged as unused.
